### DISCUSSION TITLE
Workspaces shifted off-screen after rebooting out of Windows

### DISCUSSION THREAD

**Discussion by flexxyfluxx** [i3/i3#5475] [Q&A] [answered]
Running a dual-boot setup of Win10/Arch+i3. Occasionally, my workspaces will be shifted down by about half a screen on all my monitors (there are 2). This seems to only ever happen after rebooting out of Windows; only 90% certain on that tho.

My proof that this is an i3 issue is that this never happened when I was running Arch+Gnome.

Any ideas? I don't fancy manually applying an arandr config every time I boot Arch after booting Windows. (the arandr config is being loaded automatically anyways, so idk what gives)

Currently solving this by just rebooting again (also not ideal).

**Comment by orestisfl** [comment]
By shifting you mean that your outputs actually lie outside the physical monitor space?

> My proof that this is an i3 issue is that this never happened when I was running Arch+Gnome.

That doesn't prove that it's an i3 issue, it could be that Gnome (which is a desktop environment and runs more software than a window manager by definition) handles your situation more gracefully than what your X server does by default.

You can test with a different wm instead of i3 like openbox & awesome and see if the issue persists.

This generally doesn't sound i3-specific because i3 doesn't save any state regarding outputs, the information it gets comes from randr directly.

> applying an arandr config

Another workaround could be found with [autorandr](https://github.com/phillipberndt/autorandr)

**Reply by flexxyfluxx** [reply]
Yes, that is correct. The workspaces are shifted down about half a screen, so that only the top half of the workspace is actually on the display, while the top half of the display is just garbage data. Will provide a picture next time I catch it happening.

For now, I will try if autorandr helps.

**Reply by flexxyfluxx** [reply]
Update: it happened.

![IMG_20230407_143312791.jpg](https://user-images.githubusercontent.com/82163962/230609592-a6434883-ffbc-48c7-a93c-f7d274596b52.jpg)

Funnily enough, I seem to be able to move my mouse outside the rendered area Also, the edge between rendered and not rendered on the left monitor seems to be aligned with the top edge of the right monitor when moving my mouse from monitor to monitor.

**Reply by flexxyfluxx** [reply]
Another update: While the workspaces are misaligned, the mouse is not, ie. if I click at the top of the screen, it is interpreted as at the top of the workspace, not somewhere in the ether above it. Also, it occasionally takes multiple reboots after rebooting from Windows to have displays not be misaligned.

**Reply by flexxyfluxx** [reply, marked as answer]
Further update: Solved (I think/hope) by setting a proper Xorg config in /etc/X11 (via NVIDIA settings) rather than just running an xrandr command generated by arandr in my i3 config, and coincidentally also solved the problem of my login screen appearing on the wrong monitor.

Now, my dmenu/window title font is too large.

**Reply by flexxyfluxx** [reply]
Solved font scaling issue by manually setting DPI to 96 in /etc/X11/xorg.conf.
